### PR TITLE
Setting a post to be sticky AND password protected should fail

### PR DESCRIPTION
**Pull request by joehoyle**
See #762

### CODE REVIEW

File: lib/endpoints/class-wp-json-posts-controller.php [hunk 1]
```diff
@@ -639,6 +639,20 @@ protected function prepare_item_for_database( $request ) {
 			if ( ! current_user_can( $post_type->cap->publish_posts ) ) {
 				return new WP_Error( 'json_forbidden', __( 'Sorry, you are not allowed to create password protected posts in this post type' ), array( 'status' => 403 ) );
 			}
+
+			if ( ! empty( $schema['properties']['sticky'] ) && ! empty( $request['sticky'] ) ) {
```
Comment (danielbachhuber) [inline lib/endpoints/class-wp-json-posts-controller.php]: Where is the password conditional in this statement?

Also should check that an already sticky post can't be assigned a password.

Comment (danielbachhuber) [inline lib/endpoints/class-wp-json-posts-controller.php]: Erp, nevermind :) missed the rest of the code.

File: lib/endpoints/class-wp-json-posts-controller.php [hunk 2]
```diff
@@ -639,6 +639,20 @@ protected function prepare_item_for_database( $request ) {
...
+				return new WP_Error( 'json_invalid_field', __( 'A post can not be sticky and have a password.' ), array( 'status' => 400 ) );
+			}
+
+			if ( ! empty( $prepared_post->ID ) && is_sticky( $prepared_post->ID ) ) {
```
Comment (danielbachhuber) [inline lib/endpoints/class-wp-json-posts-controller.php]: Need WordPress objects here :) `$obj->is_sticky()`

Comment (johnbillion) [inline lib/endpoints/class-wp-json-posts-controller.php]: Tell me about it

**Pull request merged**